features/admin.applications.v1/components/edit-application.tsx

```diff
@@ -222,724 +231,164 @@ export const EditApplication: FunctionComponent<EditApplicationPropsInterface> =
-                );
-
-                if(protocolTabIndex !== -1) {
-                    handleActiveTabIndexChange(protocolTabIndex);
-                }
-
-                return;
-            } else if (urlSearchParams.get(ApplicationManagementConstants.IS_ROLES) === "true") {
-                const rolesTabIndex: number = renderedTabPanes?.findIndex(
-                    (element: {"componentId": string}) =>
-                        element.componentId === ApplicationManagementConstants.ROLES_TAB_URL_FRAG
-                );
-
-                if (rolesTabIndex !== -1) {
-                    handleActiveTabIndexChange(rolesTabIndex);
+                if(!protocolNames.includes(protocolName)) {
+                    protocolNames.push(protocolName);
+                    inboundProtocolRequests.push(getInboundProtocolConfig(appId, protocolName));
                 }
+            });
 
-                return;
-            }
-            else {
-                handleDefaultTabIndexChange(defaultTabIndex);
-            }
-        }
-    },[ template, renderedTabPanes, urlSearchParams ]);
-
-    /**
-     * Check whether the application is an M2M Application.
-     */
-    useEffect(() => {
-
-        if (template?.id === ApplicationTemplateIdTypes.M2M_APPLICATION) {
-            setM2MApplication(true);
-        }
-    }, [ template ]);
-
-    /**
-     * Called when the URL fragment updates.
-     */
-    useEffect( () => {
-
-        if(totalTabs === undefined || window.location.hash.includes(URLFragmentTypes.VIEW) ||
-            isEmpty(window.location.hash)) {
-
-            return;
-        }
-
-        const urlFragment: string[] = window.location.hash.split("#"+URLFragmentTypes.TAB_INDEX);
-
-        if(urlFragment.length === 2 && isEmpty(urlFragment[0]) && /^\d+$/.test(urlFragment[1])) {
-
-            const tabIndex: number = parseInt(urlFragment[1], 10);
-
-            if(tabIndex === activeTabIndex) {
-                return;
-            }
-
-            handleActiveTabIndexChange(tabIndex);
-        } else if (window.location.hash.includes(ApplicationManagementConstants.SIGN_IN_METHOD_TAB_URL_FRAG)) {
-            // Handle loading sign-in method tab when redirecting from the "Connected Apps" Tab of an IdP.
-            const SignInMethodtabIndex: number = renderedTabPanes?.findIndex(
-                (element: {"componentId": string}) =>
-                    element.componentId === ApplicationManagementConstants.SIGN_IN_METHOD_TAB_URL_FRAG);
-
-            handleActiveTabIndexChange(SignInMethodtabIndex);
-        } else {
-            // Change the tab index to defaultActiveIndex for invalid URL fragments.
-            handleDefaultTabIndexChange(defaultActiveIndex);
-        }
-    }, [ window.location.hash, totalTabs ]);
+            setIsInboundProtocolConfigRequestLoading(true);
+            Axios.all(inboundProtocolRequests).then(Axios.spread((...responses: AxiosResponse[]) => {
```

Comment:
We can use httpRequestAll function from the react sdk instead.

Reply:
This is an existing code block. https://github.com/wso2/identity-apps/blob/05605b9367987c95ce06add63ce0e95494a90b2d/features/admin.applications.v1/components/edit-application.tsx#L848-L849